### PR TITLE
Feature/gpu mining - Make build with OpenCL as an option

### PR DESCRIPTION
**Pull request by ShengguangXiao**
## Description

Make  build with OpenCL as an option, remove the OpenCL as a compulsary dependency, 

Add a CMAKE option OPENCL_MINE

## Review Suggestion

Please check the main CMakeLists.txt and build script.

Run ./build.sh to check if build can success. And enable OPENCL_GPU_MINE option in constants.xml and run ./build/tests/POW/Test_POW, there should be error throw out say software is not build with OpenCL.

### Implementation

- [x] **ready for review**

### Integration Test (Core Team)

- [x] local machine test

- [x] small-scale cloud test

### CODE REVIEW

**Comment by Gnnng** [comment]
Seems this PR is not based from the latest master so there are some conflicts. Can you do a rebase, @ShengguangXiao?

**Comment by bb111189** [comment]
Remember to add labels and Projects --->

File: README.md [hunk 1]
```diff
@@ -102,7 +102,11 @@ To run Zilliqa, we recommend the following minimum system requirements:
     pkill zilliqa
     ``` 
 
-## Joining public testnet 
+## GPU Mining
```
Comment (Gnnng) [inline README.md]: To keep README.md simple and clean, we perhaps don't need this section for now as there are little information to offer here and the link to wiki has been added to **Available features**.

See https://github.com/Zilliqa/Zilliqa/blame/master/README.md#L44

**Comment by Gnnng** [comment]
I just did an example of rebase and pushed the result to the branch `fix/gpu_mining`.  The diff is [here](https://github.com/Zilliqa/Zilliqa/compare/fix/gpu_mining). 

You can see that the history is much clearer without the "merge conflicts" commits and later people will easily track the changes from the "linear" commit history.

**Comment by ShengguangXiao** [comment]
@Gnnng @bb111189 The rebase is done, please help to review again

**Pull request merged**